tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/amqp/connect/ValidateAMQPErrorsTest.java

```diff
@@ -0,0 +1,343 @@
+         ConnectionFactory cf2 = CFUtil.createConnectionFactory("AMQP", "tcp://localhost:" + AMQP_PORT_2);
+
+         try (Connection connection = cf1.createConnection()) {
+            Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            MessageProducer producer = session.createProducer(session.createQueue(getQueueName()));
+            for (int i = 0; i < 10; i++) {
+               producer.send(session.createTextMessage("message " + i));
+            }
+         }
+
+         // messages should still flow after a disconnect on the link
+         // the server should reconnect as if it was a failure
+         try (Connection connection = cf2.createConnection()) {
+            Session session = connection.createSession(false, Session.AUTO_ACKNOWLEDGE);
+            MessageConsumer consumer = session.createConsumer(session.createQueue(getQueueName()));
+            connection.start();
+            for (int i = 0; i < 10; i++) {
+               Assert.assertEquals("message " + i, ((TextMessage) consumer.receive(5000)).getText());
+            }
+         }
+
+      } finally {
+         AssertionLoggerHandler.stopCapture();
+      }
+   }
+
+   @Test
+   public void testTimeoutOpenOnReceiver() throws Exception {
+
+      Vertx vertx = Vertx.vertx();
+
+      ProtonServerOptions serverOptions = new ProtonServerOptions();
+      serverOptions.setPort(AMQP_PORT);
+
+      MockServer mockServer = new MockServer(vertx, serverOptions, null, serverConnection -> {
+         serverConnection.openHandler(serverSender -> {
+            serverConnection.open();
+         });
+         serverConnection.sessionOpenHandler((s) -> {
+            s.open();
+         });
+         serverConnection.senderOpenHandler((x) -> {
+            x.open();
+         });
+         serverConnection.receiverOpenHandler((x) -> {
+            //x.open(); // I'm missing the open, so it won't ever connect
+         });
+      });
+
+      try {
+         AssertionLoggerHandler.startCapture();
+         server.setIdentity("Server1");
+
+         AMQPBrokerConnectConfiguration amqpConnection = new AMQPBrokerConnectConfiguration("test", "tcp://localhost:" + mockServer.actualPort() + "?connect-timeout-millis=20").setReconnectAttempts(5).setRetryInterval(10);
+         amqpConnection.addElement(new AMQPBrokerConnectionElement().setMatchAddress(getQueueName()).setType(AMQPBrokerConnectionAddressType.SENDER));
+         amqpConnection.addElement(new AMQPMirrorBrokerConnectionElement());
+         server.getConfiguration().addAMQPConnection(amqpConnection);
+         server.start();
+
+         Wait.assertTrue(() -> AssertionLoggerHandler.findText("AMQ111001"));
+
+      } finally {
+         AssertionLoggerHandler.stopCapture();
+         mockServer.close();
+         vertx.close();
+      }
+   }
+
+   @Test
+   public void testNoCapability() throws Exception {
```

Comment:
Saying which capability would be good (clearly I know now, others may not, or people may forget later..). Also this could specifically test the broker set it as desired, and thus explicitly show where the offered capability is being omitted.